File: 370-range-addition/370-range-addition.py
class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        diff = [0] * length

        
        for u in updates:
            i, j, val = u
            diff[i] += val
            if j + 1 < length:
                diff[j + 1] -= val
        
        
        for i in range(1, length):
            diff[i] += diff[i - 1]
        return diff
                

File: 370-range-addition/370-range-addition.java
class Solution {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] diff = new int[length];
        
        for(int[] update :updates) {
            int l = update[0];
            int r = update[1];
            int val = update[2];
            diff[l] += val;
            if(r + 1 < length) diff[r + 1] -= val;
        }
        for(int i = 1; i <length; i++) {
            diff[i] += diff[i - 1];
        }
        return diff;
    }
}